scripts/generate_zoo_yml.py

```diff
@@ -0,0 +1,57 @@
+import structlog
+
+from zoo.repos.github import create_remote_commit as create_github_remote_commit
+from zoo.repos.github import get_file_content as get_github_file_content
+from zoo.repos.gitlab import create_remote_commit as create_gitlab_remote_commit
+from zoo.repos.gitlab import get_file_content as get_gitlab_file_content
+from zoo.repos.models import Repository
+from zoo.repos.zoo_yml import generate, parse, validate
+from zoo.services.models import Service
+
+log = structlog.get_logger()
+
+
+def file_exists(remote_id, path, provider, ref="master"):
+    get_file_content = None
```

Comment:
Hmm, what do you think about
```suggestion
from zoo.repos import github, gitlab
...
remotes = {
    "github": github,
    "gitlab": gitlab,
}
...
def file_exists(remote_id, path, provider, ref="master"):
    remote =  remotes[provider]
   
    try:
        remote.get_file_content(remote_id, path, ref)
        return True
    except FileNotFoundError:
        return False   
```
and avoiding those commits, which can potentially grow if we add more providers or we use more methods from them?

Reply:
Yeah I hate python for not having proper interfaces :) But yeah I forgot that modules are first class so you can store them. This approach is acceptable to me, would say at least better than what I have here.

Reply:
Django command now has get_provider method that uses the above logic and its called only in one place. All other methods that need provider now get a full provider already calculated not just the name.